shared/reducers/chat.js

```diff
@@ -289,47 +305,27 @@ function reducer(state: Constants.State = initialState, action: Constants.Action
       if (action.error) {
         console.warn('Error in updateTempMessage')
         const {conversationIDKey, outboxID} = action.payload
-
-        // $FlowIssue
-        return state.update('conversationStates', conversationStates =>
-          updateConversationMessage(
-            conversationStates,
-            conversationIDKey,
-            item => !!item.outboxID && item.outboxID === outboxID,
-            m => ({
-              ...m,
-              messageState: 'failed',
-            })
-          )
-        )
+        return updateStateWithMessageOutboxIDChanged(state, conversationIDKey, outboxID, {
+          messageState: 'failed',
+        })
       } else {
         const {outboxID, message, conversationIDKey} = action.payload
-
-        // $FlowIssue
-        return state
-          .update('conversationStates', conversationStates =>
-            updateConversationMessage(
-              conversationStates,
-              conversationIDKey,
-              item => !!item.outboxID && item.outboxID === outboxID,
-              m => ({
-                ...m,
-                ...message,
-              })
-            )
-          )
-          .update('inbox', inbox =>
-            inbox.map((i, inboxIdx) => {
-              // Update snippetKey to message.messageID so we can clear deleted message snippets
-              if (i.get('conversationIDKey') === conversationIDKey) {
-                if (i.get('snippetKey') === outboxID && message.messageID) {
-                  return i.set('snippetKey', message.messageID)
-                }
+        return updateStateWithMessageOutboxIDChanged(
+          state,
+          conversationIDKey,
+          outboxID,
+          message
+        ).update('inbox', inbox =>
+          inbox.map((i, inboxIdx) => {
+            // Update snippetKey to message.messageID so we can clear deleted message snippets
+            if (i.get('conversationIDKey') === conversationIDKey) {
+              if (i.get('snippetKey') === outboxID && message.messageID) {
+                return i.set('snippetKey', message.messageID)
               }
-              return i
-            })
-          )
-          .set('messageMap', state.get('messageMap').set(message.key, message))
```

Comment:
If `message` was a partial object (for updating the existing message) this would blow away message data.